slides.typ

```diff
@@ -94,74 +178,33 @@
 #let slides(
     title: none,
     author: none,
+    subtitle: none,
     short-title: none,
     short-author: none,
     date: none,
-    color: teal,
-    document_body
+    theme: slides-default-theme(),
+    body
 ) = {
-    show heading.where(level: 1): h => {
-        section.update(h.body)
-    }
-
-    show heading.where(level: 2): h => {
-        pagebreak(weak: true)
-        logical-slide.step()
-        h
-    }
-
     set text(
         size: 25pt,
     )
 
-    let decoration(position, body) = {
-        let border = 1mm + color
-        let strokes = (
-            header: ( bottom: border ),
-            footer: ( top: border )
-        )
-        block(
-            stroke: strokes.at(position),
-            width: 100%,
-            inset: .3em,
-            text(.5em, body)
-        )
-    }
-
     set page(
         paper: "presentation-16-9",
-        margin: ( x: 1em, y: 4em ),
-        header: locate( loc => {
-            if counter(page).at(loc).first() > 1 {
-                decoration("header", section.at(loc))
-            }
-        } ),
-        header-ascent: 3em,
-        footer: locate( loc => {
-            if counter(page).at(loc).first() > 1 {
-                decoration("footer")[
-                    #short-author #h(10fr)
-                    #short-title #h(1fr)
-                    #date #h(10fr)
-                    #logical-slide.display()
-                ]
-            }
-        } ),
-        footer-descent: 3em,
+        margin: 0pt,
     )
 
-    [
-        #align(center + horizon,
-            block(fill: color, inset: 1em, radius: 1em, breakable: false,
-                [
-                    #text(2em)[*#title*] \
-                    #v(1em)
-                    #text(1.5em)[#author] \
-                    #v(1em)
-                    #date
-                ]
-            )
-        )
-        #document_body
-    ]
+    let data = (
+        title: title,
+        author: author,
```

Comment:
I believe we should allow passing lists of authors to themes as well (that is, lists should be the type the themes receive; if the user gives a single author as parameter to `#slides`, it should be automatically wrapped in an array).

Reply:
Good idea. Then it's up to the theme if they want to do something fancy with it or just join the array back into one string.